s3path.py

```diff
@@ -887,15 +888,15 @@ def is_file(self):
         except ClientError:
             return False
 
-    def iterdir(self):
+    def iterdir(self) -> Iterator["S3Path"]:
```

Comment:
I think for this one you want `from typing import Generator` and then change this to
```suggestion
    def iterdir(self) -> Generator["S3Path", None, None]:
```

Reply:
Iterator is also allowed for functions with yield. This example comes from the python documentation itself:
```python
def infinite_stream(start: int) -> Iterator[int]:
    while True:
        yield start
        start += 1
```

https://docs.python.org/3/library/typing.html#typing.Generator

Reply:
all this time i've been over-complicating the return types of my yield-only generators. thanks for the lesson!